src/pipeline/rackhd/docker/deploy.sh

```diff
@@ -0,0 +1,370 @@
+    echo "[DEBUG] volume need rm: $volume"
+    if [ -n "$volume" ]; then
+      echo $SUDO_PASSWORD |sudo -S docker volume rm $volume
+    fi
+    echo "Clean up done."
+}
+
+##############################################
+#
+# deploy RackHD 
+#
+##############################################
+deployRackHD(){
+    local rackhd_dir=$1
+    local on_build_config_dir=$2
+    local rackhd_docker_images=$3
+    local log_dir=$4
+    local build_record_file=$5
+
+    # CleanUp & stop service
+    prepareEnv ${rackhd_dir}
+    # Customize the config.json
+    setupRackHDConfig  ${on_build_config_dir} ${rackhd_dir}
+    # Customize the docker-compose.yml
+    setupDockerComposeConfig ${rackhd_docker_images} ${rackhd_dir} ${build_record_file}  
+    # Build docker image and run it
+    dockerUp ${log_dir} ${rackhd_dir}
+    echo "Deploy RackHD done."
+}
+
+prepareEnv(){
+    echo "*****************************************************************************************************"
+    echo "Start to clean up environment: stopping running containers, service mongodb and rabbitmq-server"
+    echo "*****************************************************************************************************"
+    cleanUp $1
+    stopServices
+    echo "Prepare Env done."
+}
+
+#########################################
+#
+# Stop Host services to avoid noise, there're mongo/rabbitmq inside docker , port confliction with OS's mongo/rabbitmq will occur.
+#
+#########################################
+stopServices(){
+    echo "Stop Services (mongo/rabbitmq)..."
+    set +e
+    netstat -ntlp |grep ":27017 "
+    mongo_port_in_use=$?
+    netstat -ntlp |grep ":5672 "
+    rabbitmq_port_in_use=$?
+    if [ "$mongo_port_in_use" == "0" ]; then
+        echo $SUDO_PASSWORD |sudo -S service mongodb stop
+    fi
+    if [ "$rabbitmq_port_in_use" == "0" ]; then
+        echo $SUDO_PASSWORD |sudo -S service rabbitmq-server stop
+    fi
+    set -e
+    echo "Stop services done."
+}
+
+###################################
+#
+# Modify the RackHD default Config file
+#
+###################################
+setupRackHDConfig(){
+    local on_build_config_dir=$1
+    local rackhd_dir=$2
+
+    echo "*****************************************************************************************************"
+    echo "Customize RackHD Config Files to adopt RackHD docker enviroment"
+    echo "*****************************************************************************************************"
+    local rackhd_dhcp_host_ip=$(ifconfig | awk '/inet addr/{print substr($2,6)}' |grep 172.31.128)
+    
+    find ${on_build_config_dir} -type f -exec sed -i -e "s/172.31.128.1/${rackhd_dhcp_host_ip}/g" {} \;
+
+    # in vagrant or ova， rackhd ip are all default  172.31.128.1
+    # but for docker containers it‘s hard to virtualize such a IP
+    # so replace it with DOCKER_RACKHD_IP which is usually the eth1 IP of vmslave
```

Comment:
actually, the reason is not "hard to virtualize such a IP"
because in CI test , rackhd docker run in host-network-mode, aka, it will re-use the Host OS's NIC.
because our CI vmslave NIC IP is 172.31.128.250(instead of 172.31.128.1) (reason unknown...), so we will have to replace the southbound IP in FIT code/config with 172.31.128.250